drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/dom/textbox/TextBoxBigIntegerSingletonDOMElementFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.guided.dtable.client.widget.table.columns.dom.textbox;
+
+import java.math.BigInteger;
+
+import org.drools.workbench.screens.guided.dtable.client.widget.table.GuidedDecisionTableView;
+import org.uberfire.ext.widgets.common.client.common.NumericBigIntegerTextBox;
+import org.uberfire.ext.wires.core.grids.client.widget.IGridLayer;
+
+public class TextBoxBigIntegerSingletonDOMElementFactory extends TextBoxSingletonDOMElementFactory<BigInteger, NumericBigIntegerTextBox> {
+
+    public TextBoxBigIntegerSingletonDOMElementFactory( final IGridLayer gridLayer,
+                                                        final GuidedDecisionTableView gridWidget ) {
+        super( gridLayer,
+               gridWidget );
+    }
+
+    @Override
+    public NumericBigIntegerTextBox createWidget() {
+        return new NumericBigIntegerTextBox( true );
+    }
+
+    @Override
+    public String convert( final BigInteger value ) {
+        if ( value == null ) {
+            return "";
+        }
+        return value.toString();
+    }
+
+    @Override
+    public BigInteger convert( final String value ) {
+        return new BigInteger( value );
```

Comment:
This can in theory throw a NumberFormatException. This is caught in similar case in ListBoxByteSingletonDOMElementFactory, might be good to catch it here too.